featuretools/computational_backends/feature_set_calculator.py

```diff
@@ -932,11 +935,22 @@ def update_feature_columns(feature_data, data):
             new_cols[name] = value
 
     # Handle the case where a dict is being updated
-    if isinstance(data, dict):
-        data.update(new_cols)
-        return data
+    if isinstance(df, dict):
```

Comment:
is it a common occurrence where `df` is a `dict`.  Asking cause it feels weird to name this `df` when there are cases where it's not actually a DataFrame.

Reply:
That's fair. I didn't think through this case when I renamed previously. I will switch this back to `data`.